server/graphql/v2/mutation/CreateProjectMutation.js

```diff
@@ -0,0 +1,70 @@
+import { GraphQLNonNull } from 'graphql';
+import { pick } from 'lodash';
+
+import roles from '../../../constants/roles';
+import models from '../../../models';
+import { NotFound, Unauthorized } from '../../errors';
+import { AccountReferenceInput, fetchAccountWithReference } from '../input/AccountReferenceInput';
+import { ProjectCreateInput } from '../input/ProjectCreateInput';
+import { Project } from '../object/Project';
+
+const DEFAULT_PROJECT_SETTINGS = {};
+
+async function createProject(_, args, req) {
+  const { remoteUser } = req;
+
+  if (!remoteUser) {
+    throw new Unauthorized('You need to be logged in to create a Project');
+  }
+
+  const parent = await fetchAccountWithReference(args.parent);
+  if (!parent) {
+    throw new NotFound('Parent not found');
+  }
+  if (!req.remoteUser.hasRole([roles.ADMIN, roles.MEMBER], parent.id)) {
+    throw new Unauthorized(`You must be logged in as a member of the ${parent.slug} collective to create a Project`);
+  }
+
+  const projectData = {
+    type: 'PROJECT',
+    ...pick(args.project, ['name', 'slug', 'description']),
+    ...pick(parent, ['currency', 'HostCollectiveId', 'isActive', 'platformFeePercent', 'hostFeePercent']),
+    approvedAt: parent.isActive ? new Date() : null,
+    ParentCollectiveId: parent.id,
+    CreatedByUserId: remoteUser.id,
+    settings: { ...DEFAULT_PROJECT_SETTINGS, ...args.project.settings },
+  };
+
+  const checkSlug = await models.Collective.findOne({ where: { slug: projectData.slug.toLowerCase() } });
```

Comment:
We also need to check the `slug` with `isBlacklistedCollectiveSlug`. Because the page relies on `/${collective}/${project}`, we may also need to add some more entries in this list to avoid any conflict. But I'd argue we should even use a sub-url like `/{collective}/projects/{project}` (ie. `/my-fund/projects/my-project`) to keep it consistent with events and avoid those kind of conflicts in the future.

Related: https://github.com/opencollective/opencollective/issues/1359

Reply:
We'll use the "sub-url" pattern like Events. However, I haven't implemented the same "auto unique slug" we have in Events as I would like to reconsider that.

I will definitely add support for `isBlacklistedCollectiveSlug`. Seems, we don't use `isBlacklistedCollectiveSlug` in the new create collective flow and we should add that there too.